File: algorithm/python/BOJ10828.py
# problem : https://www.acmicpc.net/problem/1082

import sys

N = int(sys.stdin.readline())
stack = list()
answer = list()

for i in range(0, N):
    cmd = sys.stdin.readline()

    if cmd.startswith("push"):
        num = int(cmd[5:])
        stack.append(num)
    elif cmd.startswith("pop"):
        if len(stack) == 0:
            answer.append(-1)
        else:
            answer.append(stack.pop())
    elif cmd.startswith("size"):
        answer.append(len(stack))
    elif cmd.startswith("empty"):
        if len(stack) == 0:
            answer.append(1)
        else:
            answer.append(0)
    else:
        if len(stack) == 0:
            answer.append(-1)
        else:
            answer.append(stack[len(stack) - 1])

for item in answer:
    print(item)

File: algorithm/typescript/BOJ10828.ts
// https://www.acmicpc.net/problem/10828

import fs = require("fs");

const input: string[] = fs.readFileSync("/dev/stdin").toString().split("\n");
const N: number = Number(input[0]);
const stack: number[] = new Array();
const answer: number[] = new Array();

for (let i = 1; i <= N; i++) {
  const cmd: string = input[i];

  if (cmd.startsWith("push")) {
    const num: number = Number(cmd.substr(5, cmd.length - 1));
    stack.push(num);
  } else if (cmd.startsWith("pop")) {
    if (stack.length === 0) {
      answer.push(-1);
    } else {
      answer.push(stack.pop());
    }
  } else if (cmd.startsWith("size")) {
    answer.push(stack.length);
  } else if (cmd.startsWith("empty")) {
    if (stack.length === 0) {
      answer.push(1);
    } else {
      answer.push(0);
    }
  } else {
    if (stack.length === 0) {
      answer.push(-1);
    } else {
      answer.push(stack[stack.length - 1]);
    }
  }
}

console.log(answer.join("\n"));
